### PR TITLE
Spatialreg

### PR DESCRIPTION
**Pull request by gregmacfarlane**
This adds documented methods for spatial autoregressive models of class`sarlm` estimated with the `spatialreg` package. Begins from code supplied to #841, with some adjustments for style and adding confidence interval on estimates.

### CODE REVIEW

File: R/spdep-tidiers.R [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  
+  reg <- as_tibble(x$X) %>%
+    select(-1) %>%
+    dplyr::mutate(
+      y = x$y,
+      .fitted = x$fitted.values, 
+      .resid = x$residuals
+    )
+  
+  #TODO: Extend for spatial lags of variables
```
Comment (simonpcouch) [inline R/spdep-tidiers.R:105]: Thanks for the PR! Is this ready for review?

Comment (gregmacfarlane) [reply R/spdep-tidiers.R:105]: Yes, I believe so. If you need me to rebase to the recent updates I can do that first, but I think it's ready to look at.

Comment (simonpcouch) [reply R/spdep-tidiers.R:105]: That would be great! Thanks for your patience.

File: R/spdep-tidiers.R [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+#' @template title_desc_augment
+#'
+#' @template augment_NAs
+#'
+#' @inherit tidy.sarlm params examples
+#'
+#' @export
+#' @seealso [augment()]
+#' @family spatialreg tidiers
+augment.sarlm <- function(x) {
```
Comment (simonpcouch) [inline R/spdep-tidiers.R:95]: Could you add a `...` argument here for compatibility with the generic and `@template param_unused_dots` roxygen tag?

File: R/spdep-tidiers.R (hunk 1 above)
Comment (gregmacfarlane) [reply R/spdep-tidiers.R:105]: Rebased below

**Comment by simonpcouch** [comment]
Merged! Thanks.🙂

**Comment by petrhrobar** [comment]
Cool! Thank you for your work.

**Pull request merged**